cmake/toolchain/xcc/common.cmake

```
@@ -7,7 +7,29 @@ if(NOT EXISTS ${XTENSA_TOOLCHAIN_PATH})
   message(FATAL_ERROR "Nothing found at XTENSA_TOOLCHAIN_PATH: '${XTENSA_TOOLCHAIN_PATH}'")
 endif()
 
-set(TOOLCHAIN_HOME ${XTENSA_TOOLCHAIN_PATH}/$ENV{TOOLCHAIN_VER}/XtensaTools)
+zephyr_get(TOOLCHAIN_VER)
+if(DEFINED TOOLCHAIN_VER)
+  set(XTENSA_TOOLCHAIN_VER ${TOOLCHAIN_VER})
+else()
+  zephyr_get(TOOLCHAIN_VER_${NORMALIZED_BOARD_TARGET})
+  if(DEFINED TOOLCHAIN_VER_${NORMALIZED_BOARD_TARGET})
+    set(XTENSA_TOOLCHAIN_VER ${TOOLCHAIN_VER_${NORMALIZED_BOARD_TARGET}})
+  else()
+    message(FATAL "Environment variable TOOLCHAIN_VER must be set")
```

minor nit, as you use `zephyr_get()` then this can be:
```suggestion
    message(FATAL "TOOLCHAIN_VER must be set in environment or given as -DTOOLCHAIN_VER=<val>")
```

---

Done.